### PR TITLE
Add a singleton-style feature to the "todolist" example.

### PR DESCRIPTION
**Pull request by rfk**
This is a nice little example of how to use object references,
and also happens to help us test a way that we want to use them
in a real consumer.

@mhammond r?

### CODE REVIEW

File: uniffi_bindgen/src/bindings/python/templates/RustBufferStream.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def readString(self):
     # The Object type {{ object_name }}.
 
     def read{{ canonical_type_name }}(self):
-        handle = self.readU64();
+        handle = self._unpack_from(8, ">Q")
```
Comment (rfk) [inline uniffi_bindgen/src/bindings/python/templates/RustBufferStream.py:110]: The `readU64` method is only present if the component interface uses the `u64` type.

**Comment by rfk** [comment]
This worked with only one tiny tweak to the generated python! The Kotlin and Swift examples no longer compile, but, a good start!

File: uniffi_bindgen/src/interface/function.rs [hunk 2]
```diff
@@ -112,9 +112,6 @@ impl APIConverter<Function> for weedle::namespace::NamespaceMember<'_> {
 impl APIConverter<Function> for weedle::namespace::OperationNamespaceMember<'_> {
     fn convert(&self, ci: &mut ComponentInterface) -> Result<Function> {
         let return_type = ci.resolve_return_type_expression(&self.return_type)?;
-        if let Some(Type::Object(_)) = return_type {
```
Comment (mhammond) [inline uniffi_bindgen/src/interface/function.rs:115]: I'm surprised my new `coveralls` fixture didn't require this - I guess it's not living up to its name :)

**Review by mhammond — APPROVED** [review]
This is looking really nice.

**Pull request merged**